examples/web/nextjs-chat/src/app/api/momento/token/route.ts

```diff
@@ -29,3 +36,28 @@ export async function GET(_request: Request) {
   }
   throw new Error("Unable to get token from momento");
 }
+
+async function fetchTokenWithOpenAuth() {
+  return await authClient.generateAuthToken(
+    tokenPermissions,
+    tokenExpiresIn,
+  );
+}
+
+async function fetchTokenWithAuthCredentials() {
+  const envUsername = String(process.env.MOMENTO_AUTH_USERNAME);
+  const envPassword = String(process.env.MOMENTO_AUTH_PASSWORD);
+
+  if (!envUsername || !envPassword) {
+    throw new Error("Username and/or password not detected in environment variables");
+  }
+
+  if (envUsername !== "momento" || envPassword !== "\$erverless") {
```

Comment:
This is still essentially a public endpoint right? If a user deploys the next application with the correct username and password as part of the env vars, anyone can hit that endpoint and get a token? I think we would need the frontend to pass the username and password to the backend as part of the body/query params. Then inside this function here can verify that the username/password passed is correct

Reply:
Oh yeah, okay I'll implement that